packages/terra-modal/src/Portal.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { PropTypes } from 'react';
+import ReactDOM from 'react-dom';
+
+/**
+ * The <Portal/> component render its children into a div and append the div to <body/>.
+ */
+const propTypes = {
+  /**
+   * Whether or not the children of <Portal/> should be rendered
+   */
+  isOpen: PropTypes.bool,
+};
+
+const defaultProps = {
+  isOpen: false,
+};
+
+class Portal extends React.Component {
```

Comment:
the custom portal here handles open/close by a prop isOpen, while react-portal is more comprehensive, it supports other props besides:
- openByClickOn (e.g. <button/>, so user don't need to assign a callback to the button), 
- closeOnEsc (which is handled by <Modal/> in this PR)
- closeOnOutsideClick (which is handled by <Backdrop/> in this PR)
- callbacks: onOpen/onClose/onUpdate/beforeClose

Basically it depends on in which level we want to handle onRequestClose event and do we need callbacks in the portal lifecycle